### ISSUE TITLE
vcpkg_find_acquire_program(NUGET) clashes with nuget export

### ISSUE THREAD

**Issue by gix** [microsoft/vcpkg#12168]
**Describe the bug**
The nuget version used by `vcpkg export --nuget` (currently 5.5.1) is different from the version used by `vcpkg_find_acquire_program(NUGET)` (currently 4.8.1), but both seem to look for `downloads/tools/nuget.exe` first. If a port downloads the old version first, export will try to verify against the hash of the newer version and fail.

**To Reproduce**
1. Remove any downloaded nuget files (`downloads/tools/nuget*`)
2. Run `vcpkg install nugetport`
3. Run `vcpkg export nugetport --nuget --nuget-version=1.0.0 --nuget-id=test`

Sample port using NuGet:

`ports\nugetport`:
```
Source: nugetport
Version: 1.0.0
```
`ports\portfile.cmake`:
```
set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
include(vcpkg_common_functions)
vcpkg_find_acquire_program(NUGET)
file(WRITE ${CURRENT_PACKAGES_DIR}/share/nugetport/copyright "")
```

**Expected behavior**
Successful export.

**Failure logs** 
```
C:\dev\vcpkg> .\vcpkg.exe install nugetport; .\vcpkg.exe export nugetport --nuget --nuget-version=1.0.0 --nuget-id=test
Computing installation plan...
The following packages will be built and installed:
    nugetport[core]:x86-windows
Starting package 1/1: nugetport:x86-windows
Building package nugetport[core]:x86-windows...
-- Performing post-build validation
-- Performing post-build validation done
Building package nugetport[core]:x86-windows... done
Installing package nugetport[core]:x86-windows...
Installing package nugetport[core]:x86-windows... done
Elapsed time for package nugetport:x86-windows: 1.098 s

Total elapsed time: 1.121 s

The following packages are already built and will be exported:
    nugetport:x86-windows
Exporting package nugetport:x86-windows...
Packing nuget package...
A suitable version of nuget was not found (required v5.5.1). Downloading portable nuget v5.5.1...
File does not have the expected hash:
             url : [ https://dist.nuget.org/win-x86-commandline/v5.5.1/nuget.exe ]
       File path : [ C:\dev\vcpkg\downloads\nuget.exe ]
   Expected hash : [ 22ea847d8017cd977664d0b13c889cfb13c89143212899a511be217345a4e243d4d8d4099700114a11d26a087e83eb1a3e2b03bdb5e0db48f10403184cd26619 ]
     Actual hash : [ 42cb744338af8decc033a75bce5b4c4df28e102bafc45f9a8ba86d7bc010f5b43ebacae80d7b28c4f85ac900eefc2a349620ae65f27f6ca1c21c53b63b92924b ]

C:\dev\vcpkg>
```

**Comment by Voskrese** [comment]
```
vcpkg hash nuget.exe

22ea847d8017cd977664d0b13c889cfb13c89143212899a511be217345a4e243d4d8d4099700114a11d26a087e83eb1a3e2b03bdb5e0db48f10403184cd26619
```

delete `C:\dev\vcpkg\downloads\nuget.exe`

**Comment by gix** [comment]
> delete `C:\dev\vcpkg\downloads\nuget.exe`

It's placed there by `vcpkg_find_acquire_program`. Having to fiddle with the download cache between install and export is not really a nice option.

**Comment by Voskrese** [comment]
sorry
https://github.com/microsoft/vcpkg/blob/67ab6130b64cce6701c5a95ca3b9bdc4d949cb8a/scripts/cmake/vcpkg_find_acquire_program.cmake#L224

edit 

`https://dist.nuget.org/win-x86-commandline/v5.5.1/nuget.exe`

https://github.com/microsoft/vcpkg/blob/67ab6130b64cce6701c5a95ca3b9bdc4d949cb8a/scripts/cmake/vcpkg_find_acquire_program.cmake#L227

`22ea847d8017cd977664d0b13c889cfb13c89143212899a511be217345a4e243d4d8d4099700114a11d26a087e83eb1a3e2b03bdb5e0db48f10403184cd26619`

to https://github.com/microsoft/vcpkg/pulls

**Comment by ras0219** [comment]
Thank you for this issue! I believe I've fixed it with #12174 -- please let us know if you still have issues after pulling the latest!
